components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4866,10 +4866,11 @@ private APIDTO importOpenAPIDefinition(InputStream definition, String definition
             throw RestApiUtil.buildBadRequestException(errorDTO);
         }
 
-        // Only HTTP type APIs should be allowed
-        if (!APIDTO.TypeEnum.HTTP.equals(apiDTOFromProperties.getType())) {
-            throw RestApiUtil.buildBadRequestException("The API's type should only be HTTP when " +
-                    "importing an OpenAPI definition");
+        // Only HTTP or WEBHOOK type APIs should be allowed
+        if (!(APIDTO.TypeEnum.HTTP.equals(apiDTOFromProperties.getType())
+                || APIDTO.TypeEnum.WEBHOOK.equals(apiDTOFromProperties.getType()))) {
+            throw RestApiUtil.buildBadRequestException(
+                    "The API's type should only be HTTP when " + "importing an OpenAPI definition");
```

Comment:
Shall we correct this exception message as:
```java
throw RestApiUtil.buildBadRequestException(
                    "The API's type should be HTTP or Webhook when importing an OpenAPI definition");
```

Reply:
Didn't change this message because it was bit unusual to have an openapi def for a webhook right?

Reply:
Agreed!

Reply:
Added a generic message